### PR TITLE
feat:Convex, Curve, Aura adaptors support + Turbo eETH permissions

### PR DESCRIPTION
**Pull request by cbrit**
- Adds protos, ABIs and call handlers for ConvexCurveAdaptor.sol, CurveAdaptor.sol, and AuraERC4626Adaptor.sol
- Removes previous Turbo SOMM permissions
- Adds the following permissions for Turbo eETH: 
```
Allow Turbo eETH(0x9a7b4980C6F0FCaa50CD5f288Ad7038f434c692e) to perform the following actions

addAdaptorToCatalogue(0xC74fFa211A8148949a77ec1070Df7013C8D5Ce92) UniV3 Adaptor
addAdaptorToCatalogue(0x8a95BBAbb0039480F6DD90fe856c1E0c3D575aA1) Vesting Simple Adaptor
addAdaptorToCatalogue(0x94E28529f73dAD189CD0bf9D83a06572d4bFB26a) Curve Adaptor
addAdaptorToCatalogue(0x2750348A897059C45683d33A1742a3989454F7d6) Balancer Adaptor
addAdaptorToCatalogue(0x298d97494c5374e796368bCF15F0290771f6aE99) Aura Adaptor

addPositionToCatalogue(12) ERC20 weETH
addPositionToCatalogue(1000009) UniswapV3 wETH/weETH
addPositionToCatalogue(100000006) weETH Vesting Position
addPositionToCatalogue(6000002) Curve weETH/wETH
addPositionToCatalogue(7000001) Balancer wETH/weETH
addPositionToCatalogue(7500001) Balancer wETH/weETH
```
- Bumps version to 3.7.0
- Updates docker build to newer version of rust, cargo-chef, and uses debian base image to minimize the final image size.

### CODE REVIEW

File: proto/adaptors/curve/curve.proto [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    /*
+    * Allows strategist to add liquidity to Curve pairs that use the native asset.
+    *
+    * Represents function `addLiquidityETH(address pool, ERC20 lpToken, uint256[] orderedUnderlyingTokenAmounts, uint256 minLPAmount, bool useUnderlying, CurveGauge gauge, bytes4 selector)`
+    */
+    message AddLiquidityETH {
+        // Address of the Curve pool
+        string pool = 1;
+        // Address of the LP token
+        string lp_token = 2;
```
Comment (EricBolten) [inline proto/adaptors/curve/curve.proto:67]: The comment on `AddLiquidityETH` in the message definition doesn't include an ERC20 `lpToken`. Is the comment incorrect, or is it not supposed to be here?

File: proto/adaptors/aave/aave_v3_debt_token_flash_loan.proto [hunk 2]
```diff
@@ -103,6 +106,12 @@ message AaveV3DebtTokenAdaptorV1FlashLoan {
             DebtFTokenAdaptorV1Calls debt_f_token_v1_calls = 25;
             // Represents function calls to the CollateralFTokenAdaptor V1
             CollateralFTokenAdaptorV1Calls collateral_f_token_v1_calls = 26;
+            // Represents function calls for the ConvexCurveAdaptorV1
+            ConvexCurveAdaptorV1Calls convex_curve_v1_calls = 30;
+            // Represents function calls for the CurveAdaptorV1
+            CurveAdaptorV1Calls curve_v1_calls = 31;
+            // Represents function calls for the AuraERC4626AdaptorV1
+            AuraERC4626AdaptorV1Calls aura_erc4626_v1_calls = 32;
```
Comment (EricBolten) [inline proto/adaptors/aave/aave_v3_debt_token_flash_loan.proto:114]: Doesn't the Balancer adaptor also need parity with flash loan types?

File: steward/src/cellars.rs [hunk 3]
```diff
@@ -176,6 +189,10 @@ pub const ADAPTOR_VESTING_SIMPLE_V1_1_DEPLOYMENT1: &str =
     "3b98ba00f981342664969e609fb88280704ac479";
 pub const ADAPTOR_VESTING_SIMPLE_V1_1_DEPLOYMENT2: &str =
     "8a95bbabb0039480f6dd90fe856c1e0c3d575aa1";
+pub const ADAPTOR_AURA_ERC4626_ADAPTOR_V1: &str = "298d97494c5374e796368bcf15f0290771f6ae99";
```
Comment (EricBolten) [inline steward/src/cellars.rs]: nit: These were sorted alphabetically if you want to insert them in there in the right spots.

File: steward/src/cellars.rs [hunk 4]
```diff
@@ -75,8 +75,13 @@ lazy_static! {
 
 pub const ALLOWED_V2_0_SETUP_ADAPTORS: [(&str, &str); 0] = [];
 pub const ALLOWED_V2_2_CATALOGUE_ADAPTORS: [(&str, &str); 0] = [];
-pub const ALLOWED_V2_5_CATALOGUE_ADAPTORS: [(&str, &str); 1] =
-    [(CELLAR_TURBO_SOMM, ADAPTOR_VESTING_SIMPLE_V1_1_DEPLOYMENT2)];
+pub const ALLOWED_V2_5_CATALOGUE_ADAPTORS: [(&str, &str); 5] = [
+    (CELLAR_TURBO_EETH, ADAPTOR_UNIV3_V3),
```
Comment (EricBolten) [inline steward/src/cellars.rs]: Our definition of `ADAPTOR_UNIV3_V3` specifies `0x92611574ec9bc13c6137917481dab7bb7b173c9b` but Seven Seas referred to the adaptor at `0xC74fFa211A8148949a77ec1070Df7013C8D5Ce92` which we don't seem to have. Has there been another deployment of the adaptor?

File: steward_abi/src/aura_adaptor_v1.rs [hunk 5]
```diff
@@ -0,0 +1,575 @@
+pub use auraadaptorv1_mod::*;
```
Comment (EricBolten) [inline steward_abi/src/aura_adaptor_v1.rs]: Is this other codegen Aura adaptor file from a previous naming scheme?

File: steward_proto_go/steward_proto/aura.pb.go [hunk 6]
```diff
@@ -0,0 +1,344 @@
+//
```
Comment (EricBolten) [inline steward_proto_go/steward_proto/aura.pb.go]: Same here re: the generated file.

**Review by EricBolten — COMMENTED** [review]
Some comments and questions.

File: proto/adaptors/aave/aave_v3_debt_token_flash_loan.proto (hunk 2 above)
Comment (cbrit) [reply proto/adaptors/aave/aave_v3_debt_token_flash_loan.proto:114]: whoops

File: steward/src/cellars.rs [hunk 7]
```diff
@@ -189,9 +192,6 @@ pub const ADAPTOR_VESTING_SIMPLE_V1_1_DEPLOYMENT1: &str =
     "3b98ba00f981342664969e609fb88280704ac479";
 pub const ADAPTOR_VESTING_SIMPLE_V1_1_DEPLOYMENT2: &str =
     "8a95bbabb0039480f6dd90fe856c1e0c3d575aa1";
-pub const ADAPTOR_AURA_ERC4626_ADAPTOR_V1: &str = "298d97494c5374e796368bcf15f0290771f6ae99";
-pub const ADAPTOR_CURVE_ADAPTOR_V1: &str = "94e28529f73dad189cd0bf9d83a06572d4bfb26a";
-pub const ADAPTOR_CONVEX_CURVE_ADAPTOR_V1: &str = "98c44ff447c62364e3750c5e2ef8acc38391a8b0";
 pub const ADAPTOR_BALANCER_POOL_V1: &str = "2750348a897059c45683d33a1742a3989454f7d6";
```
Comment (EricBolten) [inline steward/src/cellars.rs]: nit: Balancer pool adaptor here still out of order.

File: steward/src/cellars.rs [hunk 8]
```diff
@@ -187,7 +187,10 @@ pub const ADAPTOR_MORPHO_AAVE_V3_A_TOKEN_COLLATERAL_V1: &str =
     "b46e8a03b1aafffb50f281397c57b5b87080363e";
 pub const ADAPTOR_MORPHO_AAVE_V3_DEBT_TOKEN_V1: &str = "25a61f771af9a38c10ddd93c2bbab39a88926fa9";
 pub const ADAPTOR_MORPHO_AAVE_V3_P2P_V1: &str = "4fe068caad05b82bf3f86e1f7d1a7b8bbf516111";
-pub const ADAPTOR_UNIV3_V3: &str = "92611574ec9bc13c6137917481dab7bb7b173c9b";
+// used by RYETH, RYBTC
+pub const ADAPTOR_UNIV3_V3_DEPLOYMENT_1: &str = "92611574ec9bc13c6137917481dab7bb7b173c9b";
+// used by Turbo stETH
```
Comment (EricBolten) [inline steward/src/cellars.rs]: nit: Turbo stETH and now Turbo EETH

**Review by EricBolten — COMMENTED** [review]
Two nits, but can approve on green CI.

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**